tests/integration/clientv3/metrics_test.go

```diff
@@ -335,20 +328,15 @@ func TestAllMetricsGenerated(t *testing.T) {
 }
 
 func getMetricsList(t *testing.T, url string) []string {
-	lines := getHTTPBodyAsLines(t, url)
-	metrics := make(map[string]struct{})
-	for _, line := range lines {
-		if strings.Contains(line, "{") {
-			metric := line[:strings.Index(line, "{")]
-			metrics[metric] = struct{}{}
-		} else {
-			metric := line[:strings.Index(line, " ")]
-			metrics[metric] = struct{}{}
-		}
+	data := getHTTPBodyAsBytes(t, url)
+	var parser expfmt.TextParser
+	mfs, err := parser.TextToMetricFamilies(bytes.NewReader(data))
+	if err != nil {
+		t.Errorf("Failed to parse metric families")
 	}
-	var metricList []string
-	for metric := range metrics {
-		metricList = append(metricList, metric)
+	var ms []string
+	for key := range mfs {
+		ms = append(ms, key)
```

Comment:
You can use [maps.Key(mfs)](https://pkg.go.dev/maps#Keys) directly

Reply:
Done